Octokit/Clients/MigrationsClient.cs

```diff
@@ -0,0 +1,133 @@
+        /// </remarks>
+        /// <param name="org">The organization which is migrating.</param>
+        /// <param name="id">Migration ID of the organization.</param>
+        /// <returns>A <see cref="Migration"/> object representing the state of migration.</returns>
+        public async Task<Migration> Get(string org, int id)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(org, "org");
+
+            var endpoint = ApiUrls.EnterpriseMigrationById(org, id);
+
+            return await ApiConnection.Get<Migration>(endpoint, null, AcceptHeaders.MigrationsApiPreview);
+        }
+
+        /// <summary>
+        /// Fetches the URL to a migration archive.
+        /// </summary>
+        /// <remarks>
+        /// https://developer.github.com/v3/migration/migrations/#download-a-migration-archive
+        /// </remarks>
+        /// <param name="org">The organization of which the migration was.</param>
+        /// <param name="id">The ID of the migration.</param>
+        /// <returns>URL as a string of the download link of the archive.</returns>
+        public async Task<string> GetArchive(string org, int id)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(org, "org");
+
+            var endpoint = ApiUrls.EnterpriseMigrationArchive(org, id);
+
+            return await ApiConnection.Get<string>(endpoint, null, AcceptHeaders.MigrationsApiPreview);
+        }
+
+        /// <summary>
+        /// Deletes a previous migration archive.
+        /// </summary>
+        /// <remarks>
+        /// https://developer.github.com/v3/migration/migrations/#delete-a-migration-archive
+        /// </remarks>
+        /// <param name="org">The organization of which the migration was.</param>
+        /// <param name="id">The ID of the migration.</param>
+        /// <returns></returns>
+        public Task DeleteArchive(string org, int id)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(org, "org");
+
+            var endpoint = ApiUrls.EnterpriseMigrationArchive(org, id);
+
+            return ApiConnection.Delete(endpoint);
+        }
+
+        /// <summary>
+        /// Unlocks a repository that was locked for migration.
+        /// </summary>
+        /// <remarks>
+        /// https://developer.github.com/v3/migration/migrations/#unlock-a-repository
+        /// </remarks>
+        /// <param name="org">The organization of which the migration was.</param>
+        /// <param name="id">The ID of the migration.</param>
+        /// <param name="repo">The repo to unlock.</param>
+        /// <returns></returns>
+        public Task UnlockRepository(string org, int id, string repo)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(org, "org");
+            Ensure.ArgumentNotNullOrEmptyString(repo, "repo");
+
+            var endpoint = ApiUrls.EnterpriseMigrationUnlockRepository(org, id, repo);
+
+            return ApiConnection.Delete(endpoint);
```

Comment:
Need to specify the preview accepts header here.  Currently the `ApiConnection.Delete()` doesn't have an override that allows this.  One is actually being added as part of another PR that is pending currently: #1185 so you could copy those changes to `IApiConnection` `ApiConnection` `IConnection` and `Connection` classes to get your side going (or wait until that PR is merged, then when you merge or rebase master you will pick it up)